transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -569,21 +579,40 @@ def _process_scan_event(self, response):
 
             if data[22] == 0xFF and data[23] == 0xC0 and data[24] == 0x3:
                 self._v1_scan_count += 1
-                self._parse_v1_advertisement(rssi, string_address, data)
+                info = self._parse_v1_advertisement(rssi, string_address, data)
             elif data[3] == 27 and data[4] == 0x16 and data[5] == 0xdd and data[6] == 0xfd:
                 self._v2_scan_count += 1
-                self._parse_v2_advertisement(rssi, string_address, data)
+                info, reading_time, stream, reading, \
+                    broadcast_toggle, counter, broadcast_multiplex = \
+                    self._parse_v2_advertisement(rssi, string_address, data)
             else:
                 pass # This just means the advertisement was from a non-IOTile device
         elif packet_type == 4:
             self._v1_scan_response_count += 1
-            self._parse_v1_scan_response(string_address, data)
+            info, reading_time, stream, reading = \
+                self._parse_v1_scan_response(string_address, data)
+
+        if info:
+            drop_broadcast = self._check_update_seen_broadcast(
+                sender, reading_time, stream, reading,
+                broadcast_toggle, counter=counter, channel=broadcast_multiplex)
+
+            if not self._throttle_scans or not drop_broadcast:
+                self._trigger_callback('on_scan', self.id, info, self.ExpirationTime)
+
+            # If there is a valid reading on the advertising data, broadcast it
+            if (not self._throttle_broadcast or not drop_broadcast) and \
```

Comment:
The way this logical condition is written seems more confusing than necessary.
```suggestion
            if not (self._throttle_broadcast and drop_broadcast) and \
```

Reply:
Agreed!